v2/mapping_services.proto

```diff
@@ -0,0 +1,140 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package org.packetbroker.v2;
+
+import "google/protobuf/empty.proto";
+import "packetbroker/api/v2/mapping.proto";
+
+option go_package = "go.packetbroker.org/api/v2;packetbroker";
+
+message GetDefaultGatewayVisibilityRequest {
+  // LoRa Alliance NetID of the Forwarder Member.
+  uint32 forwarder_net_id = 1;
+  // Forwarder identifier issued by the Forwarder Member.
+  string forwarder_id = 2;
+}
+
+message SetDefaultGatewayVisibilityRequest {
+  // LoRa Alliance NetID of the Forwarder Member.
+  uint32 forwarder_net_id = 1;
+  // Forwarder identifier issued by the Forwarder Member.
+  string forwarder_id = 2;
+  // New default Gateway Visibility.
+  GatewayVisibility visibility = 3;
+}
+
+message GetHomeNetworkGatewayVisibilityRequest {
+  // LoRa Alliance NetID of the Forwarder Member.
+  uint32 forwarder_net_id = 1;
+  // Forwarder identifier issued by the Forwarder Member.
+  string forwarder_id = 2;
+  // LoRa Alliance NetID of the Home Network Member.
+  uint32 home_network_net_id = 3;
+}
+
+message SetHomeNetworkGatewayVisibilityRequest {
+  // LoRa Alliance NetID of the Forwarder Member.
+  uint32 forwarder_net_id = 1;
+  // Forwarder identifier issued by the Forwarder Member.
+  string forwarder_id = 2;
+  // LoRa Alliance NetID of the Home Network Member.
+  uint32 home_network_net_id = 3;
+  // New Gateway Visibility for the Home Network Member.
+  GatewayVisibility visibility = 4;
+}
+
+message GetGatewayVisibilityResponse {
+  GatewayVisibility visibility = 1;
+}
+
+service GatewayVisibilityManager {
+  // Get the Default Gateway Visibility.
+  rpc GetDefaultVisibility(GetDefaultGatewayVisibilityRequest) returns (GetGatewayVisibilityResponse);
+  // Set the Default Gateway Visibility.
+  rpc SetDefaultVisibility(SetDefaultGatewayVisibilityRequest) returns (google.protobuf.Empty);
+  // Get the Gateway Visibility with the Home Network.
+  rpc GetHomeNetworkVisibility(GetHomeNetworkGatewayVisibilityRequest) returns (GetGatewayVisibilityResponse);
+  // Set the Gateway Visibility with the Home Network.
+  rpc SetHomeNetworkVisibility(SetHomeNetworkGatewayVisibilityRequest) returns (google.protobuf.Empty);
+}
+
+message UpdateForwarderGatewayRequest {
+  // LoRa Alliance NetID of the Forwarder Member.
+  uint32 forwarder_net_id = 1;
+  // Forwarder identifier issued by the Forwarder Member.
+  string forwarder_id = 2;
+  // Gateway to update.
+  Gateway gateway = 3;
+}
+
+service ForwarderMapper {
+  // Update a gateway.
+  rpc UpdateGateway(UpdateForwarderGatewayRequest) returns (google.protobuf.Empty);
+}
+
+message ListHomeNetworkGatewaysRequest {
+  // LoRa Alliance NetID of the Home Network Member.
+  uint32 home_network_net_id = 1;
+  // Number of items to skip for pagination.
```

Comment:
I think it's better to use `limit`+`page` instead of `limit`+`offset`. Reasoning is that with `offset`, you need to know the default `limit` of the server (if you don't specify one), whereas with `page` you don't.

Reply:
`limit` + `offset` gives flexibility to display without fixed page size. For example to "show more" (10-100). `page` feels too much of a frontend thing. Now, `offset` is just `limit*(page-1)` or anything else the clients wants.

If you have pages, the client already knows `limit`. If you don't care about pages, but want to receive the items in chunks with server defaults, the client specifies `0` for `limit` to scan.